extensions/applicationinsights-analytics-js/src/JavaScriptSDK/AnalyticsPlugin.ts

```diff
@@ -284,6 +284,13 @@ export class AnalyticsPlugin extends BaseTelemetryPlugin implements IAppInsights
                 if (doc) {
                     pageView.refUri = pageView.refUri === undefined ? doc.referrer : pageView.refUri;
                 }
+
+                // Access the performance timing object
+                const navigationEntries = window.performance.getEntriesByType("navigation") as PerformanceNavigationTiming[];
```

Comment:
Note: not every runtime has `window`, `window.performance`, the `getEntriesByType()` etc.
So this should use the helpers and check for their existence -- otherwise we will have runtime javascript errors.

eg.
```ts
let perf = getPerformance();
if (perf && perf.getEntriesByType) {
    const navigationEntries = perf.getEntriesByType("navigation");
   ...
} else {
    // We will need to handle this case as well and subtract the `duration` that we would normally have calculated from the `time`